python_modules/libraries/dagster-dg/dagster_dg/cli/plus.py

```diff
@@ -153,6 +161,207 @@ def pull_env_command(**global_options: object) -> None:
             )
 
 
+class EnvVarScope(str, Enum):
+    """Env var scopes in Dagster Plus.
+
+    local corresponds to secrets pulled with `dg plus env pull`,
+    branch corresponds to secrets set in branch deployments,
+    full corresponds to secrets set in full deployments.
+    """
+
+    LOCAL = "local"
+    BRANCH = "branch"
+    FULL = "full"
+
+
+def _secret_is_global(secret: Mapping[str, Any]) -> bool:
+    return not secret["locationNames"]
+
+
+def _get_secret_scopes(secret: Mapping[str, Any]) -> set[EnvVarScope]:
+    return {
+        s
+        for s in {
+            EnvVarScope.FULL if secret["fullDeploymentScope"] else None,
+            EnvVarScope.BRANCH if secret["allBranchDeploymentsScope"] else None,
+            EnvVarScope.LOCAL if secret["localDeploymentScope"] else None,
+        }
+        if s is not None
+    }
+
+
+@plus_group.group(name="create", cls=DgClickGroup)
+def plus_create_group():
+    """Commands for creating configuration in Dagster Plus."""
+
+
+@plus_create_group.command(name="env", cls=DgClickCommand)
+@click.argument("env_name")
+@click.argument("env_value", type=click.STRING, required=False)
+@click.option(
+    "--from-local-env",
+    is_flag=True,
+    help="Pull the environment variable value from your shell environment or project .env file.",
+)
+@click.option(
+    "--scope",
+    type=click.Choice(["full", "branch", "local"]),
+    multiple=True,
+    help="The deployment scope to set the environment variable in. Defaults to all scopes.",
+)
+@click.option(
+    "--global",
+    "global_",
+    is_flag=True,
+    help="Whether to set the environment variable at the deployment level, for all locations.",
+)
+@click.option(
+    "--no-confirm",
+    is_flag=True,
+    help="Do not confirm the creation of the environment variable, if it already exists.",
+)
```

Comment:
We should standardize on something here across all commands, maybe with a helper function I went with '-y' on the hybrid deployment PR, I feel like that's a common pattern in other CLIs?

```
@click.option(
    "-y", "--yes", "skip_confirmation_prompt", is_flag=True, help="Skip confirmation prompts."
)
```